# ITMGT 25.03 EE
# Assignment 3B

This is a data-cleaning assignment and is the second of two parts. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.

In addition, you have a new file called **studentprofiles.csv** which has first and last names plus the genders of the students.

The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called **cleanrecweekdata3b.csv** **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats3b.html**. The requirements for the summary statistics are described below. **(10 points)**

The outputs are similar to that of Assignment 3A, except that you will now use Pandas instead of Python lists.

This entire assignment (3B) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Monday, July 9, 2019** at 12:00NN.

Note: I understand you all need to review for LT 2. Think of this Assignment as practice. It's not meant to take you as long as 3A.

#### Bonus Points:
* +2 if submitted over (or before) the weekend (July 13-14)
* +2 if CSS and div tags are used (but please don't spend too much time on this because you still need to review for the Long Test)

In [2]:
## Code goes here

import pandas as pd

In [3]:
df = pd.read_csv("recweekdata.csv")

In [4]:
df

firstname      lastname birthmonth  birthday  birthyear  \
0        name1     lastname1         11        26       1998   
1        name2     lastname2        May        26       2001   
2        name3     lastname3          5        16       1997   
3        name4     lastname4        Dec         9         97   
4        name5     lastname5        Sep         5         97   
5        name6     lastname6      March         2       1999   
6        name7     lastname7         11        10         98   
7        name8     lastname8        Aug        28         99   
8        name9     lastname9       July        10         97   
9       name10    lastname10          9        14       2001   
10      name11    lastname11   December         8         98   
11      name12    lastname12          5         4         97   
12      name13    lastname13    January        28       1999   
13      name14    lastname14          1        11       1999   
14      name15    lastname15          6         6         98   
15      name16    lastname16          4        20         99   
16      name17    lastname17          1         3         97   
17      name18    lastname18        Apr         8       1997   
18      name19    lastname19         12        28       1997   
19      name20    lastname20          3        10         97   
20      name21    lastname21          7        13         99   
21      name22    lastname22          9        15       1999   
22      name23    lastname23          4        15         97   
23      name24    lastname24        Oct         1         97   
24      name25    lastname25          5         6         97   
25      name26    lastname26         12         5       1999   
26      name27    lastname27      March         4       1997   
27      name28    lastname28         11        10       2000   
28      name29    lastname29         10        10       1999   
29      name30    lastname30        May        28       1999   
...        ...           ...        ...       ...        ...   
2970  name2971  lastname2971   February        25         97   
2971  name2972  lastname2972        May        14       1998   
2972  name2973  lastname2973        Oct         9       1999   
2973  name2974  lastname2974          8        12       1999   
2974  name2975  lastname2975   November        18       1997   
2975  name2976  lastname2976       July        24       1998   
2976  name2977  lastname2977        May        26         98   
2977  name2978  lastname2978         10        26         99   
2978  name2979  lastname2979          2        13       1997   
2979  name2980  lastname2980       July         3         99   
2980  name2981  lastname2981        Feb        17       1998   
2981  name2982  lastname2982          3        28       2000   
2982  name2983  lastname2983   December         3         98   
2983  name2984  lastname2984        Nov        13       2000   
2984  name2985  lastname2985        Feb        23         99   
2985  name2986  lastname2986          3         2       1999   
2986  name2987  lastname2987        Oct         6         97   
2987  name2988  lastname2988        Feb        20       1997   
2988  name2989  lastname2989        Jan        17       1999   
2989  name2990  lastname2990        Jul        14         97   
2990  name2991  lastname2991       July        27       2000   
2991  name2992  lastname2992      March        26         99   
2992  name2993  lastname2993          4         1       2000   
2993  name2994  lastname2994      April        10         99   
2994  name2995  lastname2995        May        21         97   
2995  name2996  lastname2996      March        28         97   
2996  name2997  lastname2997        May        19       1998   
2997  name2998  lastname2998       July         3         98   
2998  name2999  lastname2999         12         2         98   
2999  name3000  lastname3000        Jul        13       1997   

                                    

In [5]:
# cleaning phonenumbers

# df.loc[:,"phonenumber"].replace(r"^0","+63",regex=True, inplace=True) # unnecessary
df.loc[:,"phonenumber"].replace(r"^917","+63 917",regex=True, inplace=True)
df.loc[:,"phonenumber"].replace(r"^918","+63 918",regex=True, inplace=True)
df.loc[:,"phonenumber"].replace(r"^927","+63 927",regex=True, inplace=True)

In [6]:
df

firstname      lastname birthmonth  birthday  birthyear  \
0        name1     lastname1         11        26       1998   
1        name2     lastname2        May        26       2001   
2        name3     lastname3          5        16       1997   
3        name4     lastname4        Dec         9         97   
4        name5     lastname5        Sep         5         97   
5        name6     lastname6      March         2       1999   
6        name7     lastname7         11        10         98   
7        name8     lastname8        Aug        28         99   
8        name9     lastname9       July        10         97   
9       name10    lastname10          9        14       2001   
10      name11    lastname11   December         8         98   
11      name12    lastname12          5         4         97   
12      name13    lastname13    January        28       1999   
13      name14    lastname14          1        11       1999   
14      name15    lastname15          6         6         98   
15      name16    lastname16          4        20         99   
16      name17    lastname17          1         3         97   
17      name18    lastname18        Apr         8       1997   
18      name19    lastname19         12        28       1997   
19      name20    lastname20          3        10         97   
20      name21    lastname21          7        13         99   
21      name22    lastname22          9        15       1999   
22      name23    lastname23          4        15         97   
23      name24    lastname24        Oct         1         97   
24      name25    lastname25          5         6         97   
25      name26    lastname26         12         5       1999   
26      name27    lastname27      March         4       1997   
27      name28    lastname28         11        10       2000   
28      name29    lastname29         10        10       1999   
29      name30    lastname30        May        28       1999   
...        ...           ...        ...       ...        ...   
2970  name2971  lastname2971   February        25         97   
2971  name2972  lastname2972        May        14       1998   
2972  name2973  lastname2973        Oct         9       1999   
2973  name2974  lastname2974          8        12       1999   
2974  name2975  lastname2975   November        18       1997   
2975  name2976  lastname2976       July        24       1998   
2976  name2977  lastname2977        May        26         98   
2977  name2978  lastname2978         10        26         99   
2978  name2979  lastname2979          2        13       1997   
2979  name2980  lastname2980       July         3         99   
2980  name2981  lastname2981        Feb        17       1998   
2981  name2982  lastname2982          3        28       2000   
2982  name2983  lastname2983   December         3         98   
2983  name2984  lastname2984        Nov        13       2000   
2984  name2985  lastname2985        Feb        23         99   
2985  name2986  lastname2986          3         2       1999   
2986  name2987  lastname2987        Oct         6         97   
2987  name2988  lastname2988        Feb        20       1997   
2988  name2989  lastname2989        Jan        17       1999   
2989  name2990  lastname2990        Jul        14         97   
2990  name2991  lastname2991       July        27       2000   
2991  name2992  lastname2992      March        26         99   
2992  name2993  lastname2993          4         1       2000   
2993  name2994  lastname2994      April        10         99   
2994  name2995  lastname2995        May        21         97   
2995  name2996  lastname2996      March        28         97   
2996  name2997  lastname2997        May        19       1998   
2997  name2998  lastname2998       July         3         98   
2998  name2999  lastname2999         12         2         98   
2999  name3000  lastname3000        Jul        13       1997   

                                    

In [7]:
# birthmonths manual

# birthmonths in numbers

df.loc[:,"birthmonth"].replace(r"^2","February",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^3","March",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^4","April",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^5","May",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^6","June",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^7","July",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^8","August",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^9","September",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^10","October",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^11","November",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^12","December",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"^1","January",regex=True, inplace=True)

# birthmonths (full)

df.loc[:,"birthmonth"].replace(r"February","February",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"March","March",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"April","April",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"May","May",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"June","June",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"July","July",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"August","August",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"September","September",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"October","October",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"November","November",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"December","December",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"January","January",regex=True, inplace=True)

# birthmonths (short)

df.loc[:,"birthmonth"].replace(r"Feb$","February",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Mar$","March",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Apr$","April",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Jun$","June",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Jul$","July",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Aug$","August",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Sep$","September",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Oct$","October",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Nov$","November",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Dec$","December",regex=True, inplace=True)
df.loc[:,"birthmonth"].replace(r"Jan$","January",regex=True, inplace=True)

In [8]:
df

firstname      lastname birthmonth  birthday  birthyear  \
0        name1     lastname1   November        26       1998   
1        name2     lastname2        May        26       2001   
2        name3     lastname3        May        16       1997   
3        name4     lastname4   December         9         97   
4        name5     lastname5  September         5         97   
5        name6     lastname6      March         2       1999   
6        name7     lastname7   November        10         98   
7        name8     lastname8     August        28         99   
8        name9     lastname9       July        10         97   
9       name10    lastname10  September        14       2001   
10      name11    lastname11   December         8         98   
11      name12    lastname12        May         4         97   
12      name13    lastname13    January        28       1999   
13      name14    lastname14    January        11       1999   
14      name15    lastname15       June         6         98   
15      name16    lastname16      April        20         99   
16      name17    lastname17    January         3         97   
17      name18    lastname18      April         8       1997   
18      name19    lastname19   December        28       1997   
19      name20    lastname20      March        10         97   
20      name21    lastname21       July        13         99   
21      name22    lastname22  September        15       1999   
22      name23    lastname23      April        15         97   
23      name24    lastname24    October         1         97   
24      name25    lastname25        May         6         97   
25      name26    lastname26   December         5       1999   
26      name27    lastname27      March         4       1997   
27      name28    lastname28   November        10       2000   
28      name29    lastname29    October        10       1999   
29      name30    lastname30        May        28       1999   
...        ...           ...        ...       ...        ...   
2970  name2971  lastname2971   February        25         97   
2971  name2972  lastname2972        May        14       1998   
2972  name2973  lastname2973    October         9       1999   
2973  name2974  lastname2974     August        12       1999   
2974  name2975  lastname2975   November        18       1997   
2975  name2976  lastname2976       July        24       1998   
2976  name2977  lastname2977        May        26         98   
2977  name2978  lastname2978    October        26         99   
2978  name2979  lastname2979   February        13       1997   
2979  name2980  lastname2980       July         3         99   
2980  name2981  lastname2981   February        17       1998   
2981  name2982  lastname2982      March        28       2000   
2982  name2983  lastname2983   December         3         98   
2983  name2984  lastname2984   November        13       2000   
2984  name2985  lastname2985   February        23         99   
2985  name2986  lastname2986      March         2       1999   
2986  name2987  lastname2987    October         6         97   
2987  name2988  lastname2988   February        20       1997   
2988  name2989  lastname2989    January        17       1999   
2989  name2990  lastname2990       July        14         97   
2990  name2991  lastname2991       July        27       2000   
2991  name2992  lastname2992      March        26         99   
2992  name2993  lastname2993      April         1       2000   
2993  name2994  lastname2994      April        10         99   
2994  name2995  lastname2995        May        21         97   
2995  name2996  lastname2996      March        28         97   
2996  name2997  lastname2997        May        19       1998   
2997  name2998  lastname2998       July         3         98   
2998  name2999  lastname2999   December         2         98   
2999  name3000  lastname3000       July        13       1997   

                                    

In [9]:
# courses (manual)

df.loc[:,"course"].replace(r"BS Legal Management$","BS Legal Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"Legal Management$","BS Legal Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"BS LM$","BS Legal Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"LM$","BS Legal Management",regex=True, inplace=True)

df.loc[:,"course"].replace(r"BS Communications Technology Management$","BS Communications Technology Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"Communications Technology Management$","BS Communications Technology Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"Communications Technology$","BS Communications Technology Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"BS CTM$","BS Communications Technology Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"CTM$","BS Communications Technology Management",regex=True, inplace=True)

df.loc[:,"course"].replace(r"BS Management Engineering$","BS Management Engineering",regex=True, inplace=True)
df.loc[:,"course"].replace(r"Management Engineering$","BS Management Engineering",regex=True, inplace=True)
df.loc[:,"course"].replace(r"BS ME$","BS Management Engineering",regex=True, inplace=True)
df.loc[:,"course"].replace(r"ME$","BS Management Engineering",regex=True, inplace=True)

df.loc[:,"course"].replace(r"BS Information Technology Entrepreneurship$","BS Information Technology Entrepreneurship",regex=True, inplace=True)
df.loc[:,"course"].replace(r"Information Technology Entrepreneurship$","BS Information Technology Entrepreneurship",regex=True, inplace=True)
df.loc[:,"course"].replace(r"BS ITE$","BS Information Technology Entrepreneurship",regex=True, inplace=True)
df.loc[:,"course"].replace(r"ITE$","BS Information Technology Entrepreneurship",regex=True, inplace=True)

df.loc[:,"course"].replace(r"BS Management$","BS Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"^Management$","BS Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"BS MGT$","BS Management",regex=True, inplace=True)
df.loc[:,"course"].replace(r"MGT$","BS Management",regex=True, inplace=True)



In [10]:
df

firstname      lastname birthmonth  birthday  birthyear  \
0        name1     lastname1   November        26       1998   
1        name2     lastname2        May        26       2001   
2        name3     lastname3        May        16       1997   
3        name4     lastname4   December         9         97   
4        name5     lastname5  September         5         97   
5        name6     lastname6      March         2       1999   
6        name7     lastname7   November        10         98   
7        name8     lastname8     August        28         99   
8        name9     lastname9       July        10         97   
9       name10    lastname10  September        14       2001   
10      name11    lastname11   December         8         98   
11      name12    lastname12        May         4         97   
12      name13    lastname13    January        28       1999   
13      name14    lastname14    January        11       1999   
14      name15    lastname15       June         6         98   
15      name16    lastname16      April        20         99   
16      name17    lastname17    January         3         97   
17      name18    lastname18      April         8       1997   
18      name19    lastname19   December        28       1997   
19      name20    lastname20      March        10         97   
20      name21    lastname21       July        13         99   
21      name22    lastname22  September        15       1999   
22      name23    lastname23      April        15         97   
23      name24    lastname24    October         1         97   
24      name25    lastname25        May         6         97   
25      name26    lastname26   December         5       1999   
26      name27    lastname27      March         4       1997   
27      name28    lastname28   November        10       2000   
28      name29    lastname29    October        10       1999   
29      name30    lastname30        May        28       1999   
...        ...           ...        ...       ...        ...   
2970  name2971  lastname2971   February        25         97   
2971  name2972  lastname2972        May        14       1998   
2972  name2973  lastname2973    October         9       1999   
2973  name2974  lastname2974     August        12       1999   
2974  name2975  lastname2975   November        18       1997   
2975  name2976  lastname2976       July        24       1998   
2976  name2977  lastname2977        May        26         98   
2977  name2978  lastname2978    October        26         99   
2978  name2979  lastname2979   February        13       1997   
2979  name2980  lastname2980       July         3         99   
2980  name2981  lastname2981   February        17       1998   
2981  name2982  lastname2982      March        28       2000   
2982  name2983  lastname2983   December         3         98   
2983  name2984  lastname2984   November        13       2000   
2984  name2985  lastname2985   February        23         99   
2985  name2986  lastname2986      March         2       1999   
2986  name2987  lastname2987    October         6         97   
2987  name2988  lastname2988   February        20       1997   
2988  name2989  lastname2989    January        17       1999   
2989  name2990  lastname2990       July        14         97   
2990  name2991  lastname2991       July        27       2000   
2991  name2992  lastname2992      March        26         99   
2992  name2993  lastname2993      April         1       2000   
2993  name2994  lastname2994      April        10         99   
2994  name2995  lastname2995        May        21         97   
2995  name2996  lastname2996      March        28         97   
2996  name2997  lastname2997        May        19       1998   
2997  name2998  lastname2998       July         3         98   
2998  name2999  lastname2999   December         2         98   
2999  name3000  lastname3000       July        13       1997   

                                    

In [11]:
df.loc[:,"birthyear"]=df.loc[:,"birthyear"].astype(str)

In [12]:
df.dtypes # check type of columns

firstname      object
lastname       object
birthmonth     object
birthday        int64
birthyear      object
course         object
phonenumber    object
dtype: object

In [13]:
# years

df.loc[:,"birthyear"].replace(r"^2001$","2001",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^2000$","2000",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^1999$","1999",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^1998$","1998",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^1997$","1997",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^99$","1999",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^98$","1998",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^97$","1997",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^01$","2001",regex=True, inplace=True)
df.loc[:,"birthyear"].replace(r"^00$","2000",regex=True, inplace=True)

In [14]:
df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26      1998   
1        name2     lastname2        May        26      2001   
2        name3     lastname3        May        16      1997   
3        name4     lastname4   December         9      1997   
4        name5     lastname5  September         5      1997   
5        name6     lastname6      March         2      1999   
6        name7     lastname7   November        10      1998   
7        name8     lastname8     August        28      1999   
8        name9     lastname9       July        10      1997   
9       name10    lastname10  September        14      2001   
10      name11    lastname11   December         8      1998   
11      name12    lastname12        May         4      1997   
12      name13    lastname13    January        28      1999   
13      name14    lastname14    January        11      1999   
14      name15    lastname15       June         6      1998   
15      name16    lastname16      April        20      1999   
16      name17    lastname17    January         3      1997   
17      name18    lastname18      April         8      1997   
18      name19    lastname19   December        28      1997   
19      name20    lastname20      March        10      1997   
20      name21    lastname21       July        13      1999   
21      name22    lastname22  September        15      1999   
22      name23    lastname23      April        15      1997   
23      name24    lastname24    October         1      1997   
24      name25    lastname25        May         6      1997   
25      name26    lastname26   December         5      1999   
26      name27    lastname27      March         4      1997   
27      name28    lastname28   November        10      2000   
28      name29    lastname29    October        10      1999   
29      name30    lastname30        May        28      1999   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25      1997   
2971  name2972  lastname2972        May        14      1998   
2972  name2973  lastname2973    October         9      1999   
2973  name2974  lastname2974     August        12      1999   
2974  name2975  lastname2975   November        18      1997   
2975  name2976  lastname2976       July        24      1998   
2976  name2977  lastname2977        May        26      1998   
2977  name2978  lastname2978    October        26      1999   
2978  name2979  lastname2979   February        13      1997   
2979  name2980  lastname2980       July         3      1999   
2980  name2981  lastname2981   February        17      1998   
2981  name2982  lastname2982      March        28      2000   
2982  name2983  lastname2983   December         3      1998   
2983  name2984  lastname2984   November        13      2000   
2984  name2985  lastname2985   February        23      1999   
2985  name2986  lastname2986      March         2      1999   
2986  name2987  lastname2987    October         6      1997   
2987  name2988  lastname2988   February        20      1997   
2988  name2989  lastname2989    January        17      1999   
2989  name2990  lastname2990       July        14      1997   
2990  name2991  lastname2991       July        27      2000   
2991  name2992  lastname2992      March        26      1999   
2992  name2993  lastname2993      April         1      2000   
2993  name2994  lastname2994      April        10      1999   
2994  name2995  lastname2995        May        21      1997   
2995  name2996  lastname2996      March        28      1997   
2996  name2997  lastname2997        May        19      1998   
2997  name2998  lastname2998       July         3      1998   
2998  name2999  lastname2999   December         2      1998   
2999  name3000  lastname3000       July        13      1997   

                                          course       phonenumber  
0                            

In [15]:
carrier = {"+63 917":"Globe","+63 918":"Smart", "+63 927":"Globe"}
carrier["+63 917"]

'Globe'

In [16]:
df.loc[:,"phonenumber"].str[0:7]

0       +63 918
1       +63 927
2       +63 918
3       +63 918
4       +63 927
5       +63 918
6       +63 917
7       +63 917
8       +63 918
9       +63 918
10      +63 927
11      +63 927
12      +63 917
13      +63 927
14      +63 917
15      +63 918
16      +63 927
17      +63 927
18      +63 927
19      +63 917
20      +63 927
21      +63 927
22      +63 917
23      +63 917
24      +63 918
25      +63 917
26      +63 927
27      +63 917
28      +63 917
29      +63 917
         ...   
2970    +63 918
2971    +63 917
2972    +63 917
2973    +63 917
2974    +63 927
2975    +63 927
2976    +63 918
2977    +63 918
2978    +63 918
2979    +63 918
2980    +63 927
2981    +63 927
2982    +63 918
2983    +63 917
2984    +63 927
2985    +63 927
2986    +63 918
2987    +63 927
2988    +63 927
2989    +63 927
2990    +63 918
2991    +63 917
2992    +63 917
2993    +63 917
2994    +63 927
2995    +63 917
2996    +63 917
2997    +63 917
2998    +63 927
2999    +63 918
Name: phonenumber, Lengt

In [17]:
df.loc[:,"phonenumber"].str[0:7].apply(lambda x: carrier[x])


0       Smart
1       Globe
2       Smart
3       Smart
4       Globe
5       Smart
6       Globe
7       Globe
8       Smart
9       Smart
10      Globe
11      Globe
12      Globe
13      Globe
14      Globe
15      Smart
16      Globe
17      Globe
18      Globe
19      Globe
20      Globe
21      Globe
22      Globe
23      Globe
24      Smart
25      Globe
26      Globe
27      Globe
28      Globe
29      Globe
        ...  
2970    Smart
2971    Globe
2972    Globe
2973    Globe
2974    Globe
2975    Globe
2976    Smart
2977    Smart
2978    Smart
2979    Smart
2980    Globe
2981    Globe
2982    Smart
2983    Globe
2984    Globe
2985    Globe
2986    Smart
2987    Globe
2988    Globe
2989    Globe
2990    Smart
2991    Globe
2992    Globe
2993    Globe
2994    Globe
2995    Globe
2996    Globe
2997    Globe
2998    Globe
2999    Smart
Name: phonenumber, Length: 3000, dtype: object

In [18]:
df["carrier"] = df.loc[:,"phonenumber"].str[0:7].apply(lambda x: carrier[x])
df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26      1998   
1        name2     lastname2        May        26      2001   
2        name3     lastname3        May        16      1997   
3        name4     lastname4   December         9      1997   
4        name5     lastname5  September         5      1997   
5        name6     lastname6      March         2      1999   
6        name7     lastname7   November        10      1998   
7        name8     lastname8     August        28      1999   
8        name9     lastname9       July        10      1997   
9       name10    lastname10  September        14      2001   
10      name11    lastname11   December         8      1998   
11      name12    lastname12        May         4      1997   
12      name13    lastname13    January        28      1999   
13      name14    lastname14    January        11      1999   
14      name15    lastname15       June         6      1998   
15      name16    lastname16      April        20      1999   
16      name17    lastname17    January         3      1997   
17      name18    lastname18      April         8      1997   
18      name19    lastname19   December        28      1997   
19      name20    lastname20      March        10      1997   
20      name21    lastname21       July        13      1999   
21      name22    lastname22  September        15      1999   
22      name23    lastname23      April        15      1997   
23      name24    lastname24    October         1      1997   
24      name25    lastname25        May         6      1997   
25      name26    lastname26   December         5      1999   
26      name27    lastname27      March         4      1997   
27      name28    lastname28   November        10      2000   
28      name29    lastname29    October        10      1999   
29      name30    lastname30        May        28      1999   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25      1997   
2971  name2972  lastname2972        May        14      1998   
2972  name2973  lastname2973    October         9      1999   
2973  name2974  lastname2974     August        12      1999   
2974  name2975  lastname2975   November        18      1997   
2975  name2976  lastname2976       July        24      1998   
2976  name2977  lastname2977        May        26      1998   
2977  name2978  lastname2978    October        26      1999   
2978  name2979  lastname2979   February        13      1997   
2979  name2980  lastname2980       July         3      1999   
2980  name2981  lastname2981   February        17      1998   
2981  name2982  lastname2982      March        28      2000   
2982  name2983  lastname2983   December         3      1998   
2983  name2984  lastname2984   November        13      2000   
2984  name2985  lastname2985   February        23      1999   
2985  name2986  lastname2986      March         2      1999   
2986  name2987  lastname2987    October         6      1997   
2987  name2988  lastname2988   February        20      1997   
2988  name2989  lastname2989    January        17      1999   
2989  name2990  lastname2990       July        14      1997   
2990  name2991  lastname2991       July        27      2000   
2991  name2992  lastname2992      March        26      1999   
2992  name2993  lastname2993      April         1      2000   
2993  name2994  lastname2994      April        10      1999   
2994  name2995  lastname2995        May        21      1997   
2995  name2996  lastname2996      March        28      1997   
2996  name2997  lastname2997        May        19      1998   
2997  name2998  lastname2998       July         3      1998   
2998  name2999  lastname2999   December         2      1998   
2999  name3000  lastname3000       July        13      1997   

                                          course       phonenumber carrier  
0                    

In [19]:
profiles_df = pd.read_csv("studentprofiles.csv")
profiles_df

firstname      lastname  gender
0        name1     lastname1  Female
1        name2     lastname2  Female
2        name3     lastname3  Female
3        name4     lastname4    Male
4        name5     lastname5  Female
5        name6     lastname6  Female
6        name7     lastname7  Female
7        name8     lastname8    Male
8        name9     lastname9    Male
9       name10    lastname10  Female
10      name11    lastname11    Male
11      name12    lastname12    Male
12      name13    lastname13  Female
13      name14    lastname14  Female
14      name15    lastname15    Male
15      name16    lastname16    Male
16      name17    lastname17  Female
17      name18    lastname18  Female
18      name19    lastname19    Male
19      name20    lastname20    Male
20      name21    lastname21  Female
21      name22    lastname22    Male
22      name23    lastname23    Male
23      name24    lastname24    Male
24      name25    lastname25    Male
25      name26    lastname26    Male
26      name27    lastname27  Female
27      name28    lastname28    Male
28      name29    lastname29  Female
29      name30    lastname30    Male
...        ...           ...     ...
2970  name2971  lastname2971  Female
2971  name2972  lastname2972    Male
2972  name2973  lastname2973  Female
2973  name2974  lastname2974  Female
2974  name2975  lastname2975  Female
2975  name2976  lastname2976  Female
2976  name2977  lastname2977  Female
2977  name2978  lastname2978  Female
2978  name2979  lastname2979    Male
2979  name2980  lastname2980    Male
2980  name2981  lastname2981  Female
2981  name2982  lastname2982    Male
2982  name2983  lastname2983    Male
2983  name2984  lastname2984    Male
2984  name2985  lastname2985  Female
2985  name2986  lastname2986  Female
2986  name2987  lastname2987  Female
2987  name2988  lastname2988  Female
2988  name2989  lastname2989    Male
2989  name2990  lastname2990  Female
2990  name2991  lastname2991  Female
2991  name2992  lastname2992    Male
2992  name2993  lastname2993  Female
2993  name2994  lastname2994    Male
2994  name2995  lastname2995    Male
2995  name2996  lastname2996  Female
2996  name2997  lastname2997    Male
2997  name2998  lastname2998  Female
2998  name2999  lastname2999    Male
2999  name3000  lastname3000    Male

[3000 rows x 3 columns]

In [20]:
new_df = pd.merge(df, profiles_df.loc[:,["lastname", "gender"]], on = 'lastname')
new_df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26      1998   
1        name2     lastname2        May        26      2001   
2        name3     lastname3        May        16      1997   
3        name4     lastname4   December         9      1997   
4        name5     lastname5  September         5      1997   
5        name6     lastname6      March         2      1999   
6        name7     lastname7   November        10      1998   
7        name8     lastname8     August        28      1999   
8        name9     lastname9       July        10      1997   
9       name10    lastname10  September        14      2001   
10      name11    lastname11   December         8      1998   
11      name12    lastname12        May         4      1997   
12      name13    lastname13    January        28      1999   
13      name14    lastname14    January        11      1999   
14      name15    lastname15       June         6      1998   
15      name16    lastname16      April        20      1999   
16      name17    lastname17    January         3      1997   
17      name18    lastname18      April         8      1997   
18      name19    lastname19   December        28      1997   
19      name20    lastname20      March        10      1997   
20      name21    lastname21       July        13      1999   
21      name22    lastname22  September        15      1999   
22      name23    lastname23      April        15      1997   
23      name24    lastname24    October         1      1997   
24      name25    lastname25        May         6      1997   
25      name26    lastname26   December         5      1999   
26      name27    lastname27      March         4      1997   
27      name28    lastname28   November        10      2000   
28      name29    lastname29    October        10      1999   
29      name30    lastname30        May        28      1999   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25      1997   
2971  name2972  lastname2972        May        14      1998   
2972  name2973  lastname2973    October         9      1999   
2973  name2974  lastname2974     August        12      1999   
2974  name2975  lastname2975   November        18      1997   
2975  name2976  lastname2976       July        24      1998   
2976  name2977  lastname2977        May        26      1998   
2977  name2978  lastname2978    October        26      1999   
2978  name2979  lastname2979   February        13      1997   
2979  name2980  lastname2980       July         3      1999   
2980  name2981  lastname2981   February        17      1998   
2981  name2982  lastname2982      March        28      2000   
2982  name2983  lastname2983   December         3      1998   
2983  name2984  lastname2984   November        13      2000   
2984  name2985  lastname2985   February        23      1999   
2985  name2986  lastname2986      March         2      1999   
2986  name2987  lastname2987    October         6      1997   
2987  name2988  lastname2988   February        20      1997   
2988  name2989  lastname2989    January        17      1999   
2989  name2990  lastname2990       July        14      1997   
2990  name2991  lastname2991       July        27      2000   
2991  name2992  lastname2992      March        26      1999   
2992  name2993  lastname2993      April         1      2000   
2993  name2994  lastname2994      April        10      1999   
2994  name2995  lastname2995        May        21      1997   
2995  name2996  lastname2996      March        28      1997   
2996  name2997  lastname2997        May        19      1998   
2997  name2998  lastname2998       July         3      1998   
2998  name2999  lastname2999   December         2      1998   
2999  name3000  lastname3000       July        13      1997   

                                          course       phonenumber carrier  \
0                   

# Required Summary Statistics

#### These were the questions from 3A:

1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

#### Additional Questions:
* How many Globe subscribers are: Male? Female?
* How many Smart subscribers are: Male? Female?

Answer these questions in an html file called **stats3b.html**. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


In [21]:
birthmonth_count = new_df.groupby(["birthmonth"])["lastname"].count()
birthmonth_count

birthmonth
April        267
August       241
December     276
February     232
January      270
July         249
June         259
March        246
May          273
November     277
October      178
September    232
Name: lastname, dtype: int64

In [22]:
birthmonth_count_messy = birthmonth_count.reset_index(name="count")
birthmonth_count_messy

birthmonth  count
0       April    267
1      August    241
2    December    276
3    February    232
4     January    270
5        July    249
6        June    259
7       March    246
8         May    273
9    November    277
10    October    178
11  September    232

In [23]:
birthmonth_count_clean = birthmonth_count_messy.reindex([4,3,7,0,8,6,5,1,11,10,9,2])
birthmonth_count_clean

birthmonth  count
4     January    270
3    February    232
7       March    246
0       April    267
8         May    273
6        June    259
5        July    249
1      August    241
11  September    232
10    October    178
9    November    277
2    December    276

In [24]:
birthday_count = new_df.groupby(["birthday"])["lastname"].count()
birthday_count

birthday
1     112
2     106
3      90
4     110
5     100
6      99
7      88
8     111
9      98
10    111
11    105
12     98
13    119
14    102
15    121
16    115
17    116
18    107
19    113
20     91
21    123
22    103
23    116
24    108
25    116
26    116
27    102
28    104
Name: lastname, dtype: int64

In [25]:
birthday_count_clean = birthday_count.reset_index(name="count")
birthday_count_clean

birthday  count
0          1    112
1          2    106
2          3     90
3          4    110
4          5    100
5          6     99
6          7     88
7          8    111
8          9     98
9         10    111
10        11    105
11        12     98
12        13    119
13        14    102
14        15    121
15        16    115
16        17    116
17        18    107
18        19    113
19        20     91
20        21    123
21        22    103
22        23    116
23        24    108
24        25    116
25        26    116
26        27    102
27        28    104

In [26]:
birthyear_count = new_df.groupby(["birthyear"])["lastname"].count()
birthyear_count

birthyear
1997    756
1998    746
1999    756
2000    380
2001    362
Name: lastname, dtype: int64

In [27]:
birthyear_count_clean = birthyear_count.reset_index(name="count")
birthyear_count_clean

birthyear  count
0      1997    756
1      1998    746
2      1999    756
3      2000    380
4      2001    362

In [28]:
course_count = new_df.groupby(["course"])["lastname"].count()
course_count

course
BS Communications Technology Management       589
BS Information Technology Entrepreneurship    590
BS Legal Management                           585
BS Management                                 629
BS Management Engineering                     607
Name: lastname, dtype: int64

In [29]:
course_count_clean = course_count.reset_index(name="count")
course_count_clean

course  count
0     BS Communications Technology Management    589
1  BS Information Technology Entrepreneurship    590
2                         BS Legal Management    585
3                               BS Management    629
4                   BS Management Engineering    607

In [30]:
network_count = new_df.groupby(["carrier"])["lastname"].count()
network_count

carrier
Globe    2045
Smart     955
Name: lastname, dtype: int64

In [31]:
network_count_clean = network_count.reset_index(name="count")
network_count_clean

carrier  count
0   Globe   2045
1   Smart    955

In [32]:
gender_network_count = new_df.groupby(["gender","carrier"])["lastname"].count()
gender_network_count

gender  carrier
Female  Globe       984
        Smart       494
Male    Globe      1061
        Smart       461
Name: lastname, dtype: int64

In [33]:
gender_network_count_clean = gender_network_count.reset_index(name="count")
gender_network_count_clean

gender carrier  count
0  Female   Globe    984
1  Female   Smart    494
2    Male   Globe   1061
3    Male   Smart    461

```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
             <tr><td>Male Globe subscribers:</td><td>{}</td></tr>
             <tr><td>Female Globe subscribers:</td><td>{}</td></tr>
             <tr><td>Male Smart subscribers:</td><td>{}</td></tr>
             <tr><td>Female Smart subscribers:</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [48]:
birthmonth_string = birthmonth_count_clean.loc[:,["birthmonth","count"]].apply(lambda x: "<tr><td></td><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthmonth_string


'<tr><td></td><td>January</td><td>270</td></tr><tr><td></td><td>February</td><td>232</td></tr><tr><td></td><td>March</td><td>246</td></tr><tr><td></td><td>April</td><td>267</td></tr><tr><td></td><td>May</td><td>273</td></tr><tr><td></td><td>June</td><td>259</td></tr><tr><td></td><td>July</td><td>249</td></tr><tr><td></td><td>August</td><td>241</td></tr><tr><td></td><td>September</td><td>232</td></tr><tr><td></td><td>October</td><td>178</td></tr><tr><td></td><td>November</td><td>277</td></tr><tr><td></td><td>December</td><td>276</td></tr>'

In [49]:
birthmonth_string_div = birthmonth_count_clean.loc[:,["birthmonth","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthmonth_string_div


'<div class="divTableRow"><div class="divTableCell">January</div><div class="divTableCell">270</div></div><div class="divTableRow"><div class="divTableCell">February</div><div class="divTableCell">232</div></div><div class="divTableRow"><div class="divTableCell">March</div><div class="divTableCell">246</div></div><div class="divTableRow"><div class="divTableCell">April</div><div class="divTableCell">267</div></div><div class="divTableRow"><div class="divTableCell">May</div><div class="divTableCell">273</div></div><div class="divTableRow"><div class="divTableCell">June</div><div class="divTableCell">259</div></div><div class="divTableRow"><div class="divTableCell">July</div><div class="divTableCell">249</div></div><div class="divTableRow"><div class="divTableCell">August</div><div class="divTableCell">241</div></div><div class="divTableRow"><div class="divTableCell">September</div><div class="divTableCell">232</div></div><div class="divTableRow"><div class="divTableCell">October</div><d

In [50]:
birthday_string = birthday_count_clean.loc[:,["birthday","count"]].apply(lambda x: "<tr><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthday_string


'<tr><td>1</td><td>112</td></tr><tr><td>2</td><td>106</td></tr><tr><td>3</td><td>90</td></tr><tr><td>4</td><td>110</td></tr><tr><td>5</td><td>100</td></tr><tr><td>6</td><td>99</td></tr><tr><td>7</td><td>88</td></tr><tr><td>8</td><td>111</td></tr><tr><td>9</td><td>98</td></tr><tr><td>10</td><td>111</td></tr><tr><td>11</td><td>105</td></tr><tr><td>12</td><td>98</td></tr><tr><td>13</td><td>119</td></tr><tr><td>14</td><td>102</td></tr><tr><td>15</td><td>121</td></tr><tr><td>16</td><td>115</td></tr><tr><td>17</td><td>116</td></tr><tr><td>18</td><td>107</td></tr><tr><td>19</td><td>113</td></tr><tr><td>20</td><td>91</td></tr><tr><td>21</td><td>123</td></tr><tr><td>22</td><td>103</td></tr><tr><td>23</td><td>116</td></tr><tr><td>24</td><td>108</td></tr><tr><td>25</td><td>116</td></tr><tr><td>26</td><td>116</td></tr><tr><td>27</td><td>102</td></tr><tr><td>28</td><td>104</td></tr>'

In [51]:
birthday_string_div = birthday_count_clean.loc[:,["birthday","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthday_string_div


'<div class="divTableRow"><div class="divTableCell">1</div><div class="divTableCell">112</div></div><div class="divTableRow"><div class="divTableCell">2</div><div class="divTableCell">106</div></div><div class="divTableRow"><div class="divTableCell">3</div><div class="divTableCell">90</div></div><div class="divTableRow"><div class="divTableCell">4</div><div class="divTableCell">110</div></div><div class="divTableRow"><div class="divTableCell">5</div><div class="divTableCell">100</div></div><div class="divTableRow"><div class="divTableCell">6</div><div class="divTableCell">99</div></div><div class="divTableRow"><div class="divTableCell">7</div><div class="divTableCell">88</div></div><div class="divTableRow"><div class="divTableCell">8</div><div class="divTableCell">111</div></div><div class="divTableRow"><div class="divTableCell">9</div><div class="divTableCell">98</div></div><div class="divTableRow"><div class="divTableCell">10</div><div class="divTableCell">111</div></div><div class="

In [52]:
birthyear_string = birthyear_count_clean.loc[:,["birthyear","count"]].apply(lambda x: "<tr><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthyear_string


'<tr><td>1997</td><td>756</td></tr><tr><td>1998</td><td>746</td></tr><tr><td>1999</td><td>756</td></tr><tr><td>2000</td><td>380</td></tr><tr><td>2001</td><td>362</td></tr>'

In [53]:
birthyear_string_div = birthyear_count_clean.loc[:,["birthyear","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
birthyear_string_div


'<div class="divTableRow"><div class="divTableCell">1997</div><div class="divTableCell">756</div></div><div class="divTableRow"><div class="divTableCell">1998</div><div class="divTableCell">746</div></div><div class="divTableRow"><div class="divTableCell">1999</div><div class="divTableCell">756</div></div><div class="divTableRow"><div class="divTableCell">2000</div><div class="divTableCell">380</div></div><div class="divTableRow"><div class="divTableCell">2001</div><div class="divTableCell">362</div></div>'

In [54]:
course_string = course_count_clean.loc[:,["course","count"]].apply(lambda x: "<tr><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
course_string


'<tr><td>BS Communications Technology Management</td><td>589</td></tr><tr><td>BS Information Technology Entrepreneurship</td><td>590</td></tr><tr><td>BS Legal Management</td><td>585</td></tr><tr><td>BS Management</td><td>629</td></tr><tr><td>BS Management Engineering</td><td>607</td></tr>'

In [55]:
course_string_div = course_count_clean.loc[:,["course","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
course_string_div


'<div class="divTableRow"><div class="divTableCell">BS Communications Technology Management</div><div class="divTableCell">589</div></div><div class="divTableRow"><div class="divTableCell">BS Information Technology Entrepreneurship</div><div class="divTableCell">590</div></div><div class="divTableRow"><div class="divTableCell">BS Legal Management</div><div class="divTableCell">585</div></div><div class="divTableRow"><div class="divTableCell">BS Management</div><div class="divTableCell">629</div></div><div class="divTableRow"><div class="divTableCell">BS Management Engineering</div><div class="divTableCell">607</div></div>'

In [56]:
network_string = network_count_clean.loc[:,["carrier","count"]].apply(lambda x: "<tr><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
network_string


'<tr><td>Globe</td><td>2045</td></tr><tr><td>Smart</td><td>955</td></tr>'

In [60]:
network_string_div = network_count_clean.loc[:,["carrier","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1])), axis=1).str.cat(sep="")
network_string_div


'<div class="divTableRow"><div class="divTableCell">Globe</div><div class="divTableCell">2045</div></div><div class="divTableRow"><div class="divTableCell">Smart</div><div class="divTableCell">955</div></div>'

In [58]:
gender_network_string = gender_network_count_clean.loc[:,["gender","carrier","count"]].apply(lambda x: "<tr><td>{}</td><td>{}</td><td>{}</td></tr>".format(str(x[0]),str(x[1]),str(x[2])), axis=1).str.cat(sep="")
gender_network_string


'<tr><td>Female</td><td>Globe</td><td>984</td></tr><tr><td>Female</td><td>Smart</td><td>494</td></tr><tr><td>Male</td><td>Globe</td><td>1061</td></tr><tr><td>Male</td><td>Smart</td><td>461</td></tr>'

In [61]:
gender_network_string_div = gender_network_count_clean.loc[:,["gender","carrier","count"]].apply(lambda x: """<div class="divTableRow"><div class="divTableCell">{}</div><div class="divTableCell">{}</div><div class="divTableCell">{}</div></div>""".format(str(x[0]),str(x[1]), str(x[2])), axis=1).str.cat(sep="")
gender_network_string_div


'<div class="divTableRow"><div class="divTableCell">Female</div><div class="divTableCell">Globe</div><div class="divTableCell">984</div></div><div class="divTableRow"><div class="divTableCell">Female</div><div class="divTableCell">Smart</div><div class="divTableCell">494</div></div><div class="divTableRow"><div class="divTableCell">Male</div><div class="divTableCell">Globe</div><div class="divTableCell">1061</div></div><div class="divTableRow"><div class="divTableCell">Male</div><div class="divTableCell">Smart</div><div class="divTableCell">461</div></div>'

In [66]:
export_csv = new_df.to_csv('cleanrecweekdata3b.csv')
                           

In [51]:
myhtml = """
<html>
    <body>
         <h2>Summary Statistics</h2>
         <p></p>
         <h4>People born in each Month</h4>
         <table>
             <tr><th>Month</th><th>Frequency</th></tr>
             {}
         </table>
         <p></p>
         <h4>People born on each Day</h4>
         <table>
             <tr><th>Day</th><th>Frequency</th></tr>
             {}
         </table>
         <p></p>
         <h4>People born in each Year</h4>
         <table>
             <tr><th>Year</th><th>Frequency</th></tr>
             {}
         </table>
         <p></p>
         <h4>People coming from each Course</h4>
         <table>
             <tr><th>Course</th><th>Frequency</th></tr>
             {}
         </table>
         <p></p>
         <h4>Phone Number: Globe or Smart</h4>
         <table>
             <tr><th>Carrier</th><th>Frequency</th></tr>
             {}
         </table>
         <p></p>
         <h4>Gender and Network Carrier</h4>
         <table>
             <tr><th>Gender</th><th>Carrier</th><th>Frequency</th></tr>
             {}
         </table>
    </body>
</html>
"""

In [52]:
from IPython.core.display import HTML

html_output = myhtml.format(birthmonth_string, birthday_string, birthyear_string, course_string, network_string, gender_network_string)

with open('stats3b.html',"w+") as file:
    file.write(html_output)

```

myhtml_3 = """
    <html>
        <head>
            <link rel="stylesheet" type="css" href="paris3.css" />
        </head>
        <body>
            <div id="FontA">Summary Statistics</div>
            <h2></h2>
            <div id="FontB">birthmonth tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthmonth</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">January</div><div class="divTableCell">270</div></div>
            <div class="divTableRow">
            <div class="divTableCell">February</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">March</div><div class="divTableCell">246</div></div>
            <div class="divTableRow">
            <div class="divTableCell">April</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">May</div><div class="divTableCell">273</div></div>
            <div class="divTableRow">
            <div class="divTableCell">June</div><div class="divTableCell">259</div></div>
            <div class="divTableRow">
            <div class="divTableCell">July</div><div class="divTableCell">249</div></div>
            <div class="divTableRow">
            <div class="divTableCell">August</div><div class="divTableCell">241</div></div>
            <div class="divTableRow">
            <div class="divTableCell">September</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">October</div><div class="divTableCell">178</div></div>
            <div class="divTableRow">
            <div class="divTableCell">November</div><div class="divTableCell">277</div></div>
            <div class="divTableRow">
            <div class="divTableCell">December</div><div class="divTableCell">276</div></div>
            </div>
            <h2></h2>
            <div id="FontB">birthday tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Day</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">1</div><div class="divTableCell">112</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2</div><div class="divTableCell">106</div></div>
            <div class="divTableRow">
            <div class="divTableCell">3</div><div class="divTableCell">90</div></div>
            <div class="divTableRow">
            <div class="divTableCell">4</div><div class="divTableCell">110</div></div>
            <div class="divTableRow">
            <div class="divTableCell">5</div><div class="divTableCell">100</div></div>
            <div class="divTableRow">
            <div class="divTableCell">6</div><div class="divTableCell">99</div></div>
            <div class="divTableRow">
            <div class="divTableCell">7</div><div class="divTableCell">88</div></div>
            <div class="divTableRow">
            <div class="divTableCell">8</div><div class="divTableCell">111</div></div>
            <div class="divTableRow">
            <div class="divTableCell">9</div><div class="divTableCell">98</div></div>
            <div class="divTableRow">
            <div class="divTableCell">10</div><div class="divTableCell">111</div></div>
            <div class="divTableRow">
            <div class="divTableCell">11</div><div class="divTableCell">105</div></div>
            <div class="divTableRow">
            <div class="divTableCell">12</div><div class="divTableCell">98</div></div>
            <div class="divTableRow">
            <div class="divTableCell">13</div><div class="divTableCell">119</div></div>
            <div class="divTableRow">
            <div class="divTableCell">14</div><div class="divTableCell">102</div></div>
            <div class="divTableRow">
            <div class="divTableCell">15</div><div class="divTableCell">121</div></div>
            <div class="divTableRow">
            <div class="divTableCell">16</div><div class="divTableCell">115</div></div>
            <div class="divTableRow">
            <div class="divTableCell">17</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">18</div><div class="divTableCell">107</div></div>
            <div class="divTableRow">
            <div class="divTableCell">19</div><div class="divTableCell">113</div></div>
            <div class="divTableRow">
            <div class="divTableCell">20</div><div class="divTableCell">91</div></div>
            <div class="divTableRow">
            <div class="divTableCell">21</div><div class="divTableCell">123</div></div>
            <div class="divTableRow">
            <div class="divTableCell">22</div><div class="divTableCell">103</div></div>
            <div class="divTableRow">
            <div class="divTableCell">23</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">24</div><div class="divTableCell">108</div></div>
            <div class="divTableRow">
            <div class="divTableCell">25</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">26</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">27</div><div class="divTableCell">102</div></div>
            <div class="divTableRow">
            <div class="divTableCell">28</div><div class="divTableCell">104</div></div>
            </div>
            <h2></h2>
            <div id="FontB">birthyear tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthyear</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">1997</div><div class="divTableCell">756</div></div>
            <div class="divTableRow">
            <div class="divTableCell">1998</div><div class="divTableCell">746</div></div>
            <div class="divTableRow">
            <div class="divTableCell">1999</div><div class="divTableCell">756</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2000</div><div class="divTableCell">380</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2001</div><div class="divTableCell">362</div></div>
            <h2></h2>
            <div id="FontB">course tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Course</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">BS Legal Management</div><div class="divTableCell">585</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Management</div><div class="divTableCell">629</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Communications Technology</div><div class="divTableCell">589</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Information Technology Entrepreneurship</div><div class="divTableCell">590</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Management Engineering</div><div class="divTableCell">607</div></div>
            <h2></h2>
            <div id="FontB">network tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Network</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">Globe</div><div class="divTableCell">2045</div></div>
            <div class="divTableRow">
            <div class="divTableCell">Smart</div><div class="divTableCell">955</div></div>
            </div>
            <h2></h2>
            <div id="FontB">gender and network tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Gender</div>
            <div class="divTableHead">Network</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">Female</div><div class="divTableCell">Globe</div><div class="divTableCell">984</div></div>
            <div class="divTableRow">
            <div class="divTableCell">Female</div><div class="divTableCell">Smart</div><div class="divTableCell">494</div></div>
            <div class="divTableRow">
            <div class="divTableCell">Male</div><div class="divTableCell">Globe</div><div class="divTableCell">1061</div></div>
            <div class="divTableRow">
            <div class="divTableCell">Male</div><div class="divTableCell">Smart</div><div class="divTableCell">461</div></div>
            </div>
        </body>
    </html>
"""

with open('stats_NEW_div_2.html',"w+") as file:
    file.write(myhtml_3)

```

In [67]:
# open using Safari browser

myhtml_NEW = """
    <html>
        <head>
            <link rel="stylesheet" type="css" href="paris3.css" />
        </head>
        <body>
            <div id="FontA">Summary Statistics</div>
            <h2></h2>
            <div id="FontB">birthmonth tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthmonth</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
            <h2></h2>
            <div id="FontB">birthday tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Day</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
            <h2></h2>
            <div id="FontB">birthyear tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthyear</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
            <h2></h2>
            <div id="FontB">course tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Course</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
            <h2></h2>
            <div id="FontB">network tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Network</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
            <h2></h2>
            <div id="FontB">gender and network tally</div>
            <div class="divTable parisTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Gender</div>
            <div class="divTableHead">Network</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            {}
        </body>
    </html>
"""

In [68]:
html_output2 = myhtml_NEW.format(birthmonth_string_div, birthday_string_div, birthyear_string_div, course_string_div, network_string_div, gender_network_string_div)

with open('stats3b_div.html',"w+") as file:
    file.write(html_output2)